comit/src/network/protocols/orderbook/order.rs

```diff
@@ -60,6 +61,22 @@ impl Order {
     }
 }
 
+// Since we only support a single trading pair this struct is actually
+// not needed, the information is implicit in the Order struct. Keep
+// this and the calls to order.tp().topic() to make it explicit that
+// there is only a single trading pair and the trading pair is
+// defined by the order struct.
+#[derive(Clone, Copy, Debug, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub enum TradingPair {
+    BtcDai,
+}
+
+impl TradingPair {
+    pub fn topic(&self) -> Topic {
```

Comment:
```suggestion
    pub fn to_topic(&self) -> Topic {
```

`to_XYZ` would convey the allocation that is happening as part of this function call.

Reply:
Whats the convention for consuming self vs not consuming self? I thought `to_xyz` functions consumed self. Would definitely like to convey the allocation though.

Reply:
`std::string::ToString` takes `&self` and returns `String` implying that your suggestion is correct. Is this the canonical trait for such things?

Reply:
> Whats the convention for consuming self vs not consuming self? I thought `to_xyz` functions consumed self. Would definitely like to convey the allocation though.

If self is consumed, then the method should be called `into_xyz`. See here: https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv